okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/UrlSelectorImpl.java

```diff
@@ -98,7 +98,9 @@ private static String switchWsToHttp(String url) {
 
     @Override
     public Optional<HttpUrl> redirectToNext(HttpUrl current) {
-        int index = currentUrl.updateAndGet(operand -> (operand + 1) % baseUrls.size());
+        Optional<Integer> currentIndex = indexFor(current);
+        int index = currentUrl.updateAndGet(
+                (currentUrlIndex) -> (currentIndex.orElse(currentUrlIndex) + 1) % baseUrls.size());
```

Comment:
also needs deterministic tests

Reply:
pushed deterministic test for UrlSelectorImpl